### PR TITLE
Feature: Select preferred church from a list (KIDS-1447)

### PR DESCRIPTION
**Pull request by Daniela510**
## Release Notes

- **New Features**
  - Introduced a new page for selecting a preferred church with search functionality.
  - Added error handling dialogs for failed church selection operations.
  - Enhanced user interaction by integrating user-specific data in the church selection process.
  - Added new analytics events to track user interactions during church selection.

- **Bug Fixes**
  - Improved logic for setting and retrieving the preferred church.

- **Chores**
  - Streamlined the data handling related to the preferred church in the repository.

- **Documentation**
  - Updated method signatures and functionality descriptions for clarity.

### CODE REVIEW

File: lib/features/family/features/preferred_church/preferred_church_selection_page.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                )
+              else
+                const Center(
+                  child: CustomCircularProgressIndicator(),
+                ),
+            ],
+          );
+        },
+      ),
+      floatingActionButton: Padding(
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/family/features/preferred_church/preferred_church_selection_page.dart:98]: Alternative then is a stack

File: lib/features/family/features/profiles/screens/profile_selection_screen.dart [hunk 2]
```diff
@@ -326,21 +331,46 @@ class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
...
       subtitle: "Let's link your church to make giving easier",
       buttons: [
         FunButton(
           text: 'Continue',
-          onTap: () {
-            // TODO: kids-1447 - navigate to church selection
-            context.pop(); //for now
+          onTap: () async {
+            context.pop(); // close modal
+            getIt<OrganisationBloc>().add(
```
Comment (TammiLion) [inline lib/features/family/features/profiles/screens/profile_selection_screen.dart:347]: I think we can move all of this to the PreferredChurchSelectionPage. Which would make it easier to reuse from different places. We can huddle on it tomorrow!

File: lib/features/family/features/impact_groups/widgets/dialogs/preferred_church_outcome_dialog.dart [hunk 3]
```diff
@@ -21,3 +22,22 @@ Future<void> showPreferredChurchSuccessDialog(
...
+Future<void> showPreferredChurchErrorDialog(
+  BuildContext context,
+) async {
+  return FunModal.showAsDialog(
+    context,
+    title: 'Oops, something went wrong...',
+    buttons: [
+      FunButton(
+        text: 'Retry',
+        onTap: () => context.pop(),
```
Comment (TammiLion) [inline lib/features/family/features/impact_groups/widgets/dialogs/preferred_church_outcome_dialog.dart:35]: Just to be sure I don't see any retry happening in the logic. I assume this means this dialog is shown on the selectionpage itself and you need to click the green button again?

**Review by TammiLion — APPROVED** [review]

File: lib/core/network/api_service.dart [hunk 4]
```diff
@@ -1026,6 +1026,31 @@ class APIService {
...
+      throw GivtServerFailure(
+        statusCode: response.statusCode,
+        body: response.body.isNotEmpty
+            ? jsonDecode(response.body) as Map<String, dynamic>
+            : null,
+      );
+    }
+
+    return response.statusCode == 200;
+  }
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/core/network/api_service.dart:1052]: no

File: lib/features/family/features/profiles/screens/profile_selection_screen.dart [hunk 5]
```diff
@@ -326,21 +331,46 @@ class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
...
+                    await showPreferredChurchSuccessDialog(
+                      context,
+                      collectGroup.orgName,
+                    );
+                  } else {
+                    await showPreferredChurchErrorDialog(context);
+                  }
+                },
+              ).toRoute(context),
+            );
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/family/features/profiles/screens/profile_selection_screen.dart]: no

File: lib/features/family/features/profiles/screens/profile_selection_screen.dart [hunk 6]
```diff
@@ -326,21 +331,46 @@ class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
...
+              ),
+            );
+            await Navigator.push(
+              context,
+              PreferredChurchSelectionPage(
+                onTap: (collectGroup) async {
+                  final success =
+                      await context.read<ProfilesCubit>().setPreferredChurch(
+                            collectGroup.nameSpace,
+                          );
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/family/features/profiles/screens/profile_selection_screen.dart]: no

**Pull request merged**